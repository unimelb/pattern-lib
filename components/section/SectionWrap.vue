<template>
  <component
    :is="progressive ? 'progressive-background' : element"
    :no-ratio="progressive"
    :src="progressive ? bgImage : false"
    :class="classes"
    :style="{ backgroundImage: bgImage && !progressive ? `url(${bgImage})` : '' }"
    class="section">
    <div
      :class="{
        'section__inner--sml': small,
        'section__inner--mid': mid,
        'section__inner--short': short,
        'section__inner--nopad': nopad,
      }"
      class="section__inner">
      <slot />
    </div>
  </component>
</template>

<script>
export default {
  props: {
    short: {
      type: Boolean,
      default: false,
    },
    element: {
      type: String,
      default: 'div',
    },
    nopad: {
      type: Boolean,
      default: false,
    },
    small: {
      type: Boolean,
      default: false,
    },
    mid: {
      type: Boolean,
      default: false,
    },
    bgImage: {
      type: String,
      default: '',
    },
    bgColor: {
      type: String,
      default: '',
    },
    centred: {
      type: Boolean,
      default: false,
    },
    progressive: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    classes() {
      return {
        [`bg-${this.bgColor}`]: !!this.bgColor,
        'section--centred': this.centred,
        'section--image': this.bgImage && this.bgImage.length,
      };
    },
  },
};
</script>
