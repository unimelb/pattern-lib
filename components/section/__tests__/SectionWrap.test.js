import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import SectionWrap from '../SectionWrap.vue';

expect.extend(toHaveNoViolations);

describe('SectionWrap', () => {
  it('should match snapshot', () => {
    const result = shallow(SectionWrap).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(SectionWrap).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
