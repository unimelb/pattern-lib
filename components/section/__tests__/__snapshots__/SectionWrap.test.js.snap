// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionWrap should match snapshot 1`] = `
<div
  class="section"
>
  <div
    class="section__inner"
  />
</div>
`;
